src/common/options/mds.yaml.in

```diff
@@ -1450,3 +1450,13 @@ options:
   - mds
   flags:
   - runtime
+- name: mds_extraordinary_events_dump_interval
+  type: float
+  level: advanced
+  desc: interval in seconds for dumping the recent in-memory logs when there is an extra-ordinary event.
+  long_desc: the minimum time between the extra-ordinary events where the MDS will dump the in-memory logs.
+    Set to ``0`` to disable.
+  default: 60
+  min: 0
+  services:
+  - mds
```

Comment:
Maybe this should allow to change in runtime, though when the `flags` is empty it will allow us to set it in runtime, more detail please see https://github.com/ceph/ceph/pull/45614/commits/67524b14ebfbe21ebcffff7dc83f5e7a755ead80, but it worth to add it explicitly here.

```
     flags:
     - runtime
```

Reply:
I haven't considered user input value for `mds_extraordinary_events_dump_interval` yet. I did this because the tracker ticket asks for it. Any way, the qa test is setting a value for it now. Does this tick interval need to be configurable at all? I think just using a constant of 1 second would better. Any thoughts?

Reply:
From the tracker, this will make sense only when in  low debugging case IMO, or this will introduce many redundancy logs ?

If this make sense I think you should also check the loglevel in MDS and then determine should really enable this when `mds_extraordinary_events_dump_interval` enabled ?

Reply:
@lxbsz 
From the tracker, enabling `mds_extraordinary_events_dump_interval` helps to debug as the production clusters are with 1/5 log level. So what's the log level we should avoid enabling `mds_extraordinary_events_dump_interval`? 

`debug mds = 20`? or anything higher than 1/5?

Reply:
Checked the code, IMO if the `debug mds >= 15` should be okay.

Reply:
@lxbsz 
I could read `debug_mds` value in the constructor. But I couldn't get notifications in `MDSRankDispatcher::handle_conf_change` when `debug_mds` value changes. Any idea to resolve this?

Reply:
Summary of my discussion with Xiubo:

There are two approaches to this:
1. Enable `mds_extraordinary_events_dump_interval` by default. 
Then the issue is that we couldn't get track the changes of `debug_mds` in MDSRankDispatcher class. The benefit of this approach is that the customer doesn't have to enable it manually to get the inmemory debug logs. The downside is the logs get piled up unnecessarily, even if there are no issues in the cluster.

2. Disable `mds_extraordinary_events_dump_interval` by default. 
This way `mds_extraordinary_events_dump_interval` is already tracked in MDSRankDispatcher and we just need to check if `debug_mds` is <15 or not in MDSRankDispatcher::handle_conf_change(). The benefit of this approach is that the in-memory logs would generate only when the user asks for it and the logs won't get piled up unnecessarily. The downside is the user must enable this option manually.

We have decided to go with (2) as the customer must know how this feature works and enable it themselves when needed. This seems a better approach considering the size of the logs it would generate.

Reply:
Here you are using the first approach instead, which is better.

Reply:
> Here you are using the first approach instead, which is better.

Sorry, misreaded it. You are using the second approach.